lib/library.rb

```diff
@@ -0,0 +1,60 @@
+class Library
+  attr_reader :name,
+              :library_books,
+              :authors,
+              :checked_out_books
+
+  def initialize(name)
+    @name = name
+    @library_books = []
+    @authors = []
+    @checked_out_books = []
+  end
+
+  def add_author(author)
+    @authors << author
+    @library_books << author.books
+    @library_books.flatten
+  end
+
+  def publication_time_frame_for(author)
+    time_frame = {}
+    years = author.books.map do |book|
+      book.year.to_i
+    end
+    time_frame[:start] = years.min.to_s
```

Comment:
Great job! Here though, we can call .min and .max without changing it to an integer. :)